cpu/esp32/periph/rtt_hw_rtc.c

```diff
@@ -77,27 +83,34 @@ static void _rtc_poweron(void)
     intr_matrix_set(PRO_CPU_NUM, ETS_RTC_CORE_INTR_SOURCE, CPU_INUM_RTC);
 
     /* set interrupt handler and enable the CPU interrupt */
-    xt_set_interrupt_handler(CPU_INUM_RTC, _rtc_isr, NULL);
-    xt_ints_on(BIT(CPU_INUM_RTC));
+    intr_cntrl_ll_set_int_handler(CPU_INUM_RTC, _rtc_isr, NULL);
+    intr_cntrl_ll_enable_interrupts(BIT(CPU_INUM_RTC));
 }
 
 static void _rtc_poweroff(void)
 {
     /* reset interrupt handler and disable the CPU interrupt */
-    xt_ints_off(BIT(CPU_INUM_RTC));
-    xt_set_interrupt_handler(CPU_INUM_RTC, NULL, NULL);
+    intr_cntrl_ll_disable_interrupts(BIT(CPU_INUM_RTC));
+    intr_cntrl_ll_set_int_handler(CPU_INUM_RTC, NULL, NULL);
 }
 
 uint64_t _rtc_get_counter(void)
 {
     /* trigger timer register update */
     RTCCNTL.time_update.update = 1;
+#if defined(MCU_ESP32)
```

Comment:
Not `#ifdef __xtensa__` as below?

Reply:
Yes, it requires a special handling for different ESP32x variants here because of different register structures and different register names. The `#ifdef __xtensa__` below is necessary for different platform handling where the register name is the same for all ESP32x variants.